client/landing/gutenboarding/onboarding-block/domains/index.tsx

```diff
@@ -74,6 +82,27 @@ const DomainsStep: React.FunctionComponent< Props > = ( { isModal } ) => {
 			goNext();
 		}
 	};
+	const handleDomainAvailabilityCheck = async () => {
+		if ( domain ) {
+			if ( domain?.is_free ) {
+				// is this a reliable way to check for .wordpress.com subdomains?
```

Comment:
Is this a sufficient way to check the user has selected a `.wordpress.com`? Is there an instant where a domain can be free and potentially unavailable?

Reply:
It is safe to assume that all wordpress subdomains are available.

In `generate_multiple_wpcom_suggestions()` inside `public.api/rest/wpcom-json-endpoints/class.wpcom-store-domains-api-endpoints.php`, I can see that subdomains go through a `url_is_available()` check, which is pointing to `wp-content/mu-plugins/misc.php`, and that function calls `wpmu_validate_blog_signup()` which is inside `wp-includes/ms-functions.php`, and that's where the checking is done.